### PR TITLE
IPv6 Backends

### PR DESCRIPTION
**Pull request by krawthekrow**
Test for #38. Made a separate PR since this depends on #43 to make it easier to swap out backends. The only difference is the most recent commit, which only makes changes to testing files.

### CODE REVIEW

File: forward/testing/backend_tracker.lua [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      table.insert(self.backends, b)
+   end
+   C.usleep(1000)
+end
+
+function BackendTracker:clear_backends()
+   for _, b in ipairs(self.backends) do
+      godefs.RemoveBackend(b.name)
+   end
+   C.usleep(10000)
```
Comment (cryslith) [inline forward/testing/backend_tracker.lua:27]: Why are these sleeps necessary?

File: forward/testing/unit_tests.lua [hunk 2]
```diff
@@ -34,26 +37,50 @@ end
...
 
+   local err = nil
    local test_start_time = os.clock()
+   local flush_counter = -1
    engine.main({done = function()
-      assert(os.clock() - test_start_time < 0.1,
-         "Test timed out. Possibly too many packets were dropped.")
-      return #self.out_collect_app.packets ==
+      if os.clock() - test_start_time > 0.1 then
+         err = "Test timed out. Possibly too many packets were dropped."
```
Comment (cryslith) [inline forward/testing/unit_tests.lua:54]: What causes this to happen?

File: forward/testing/backend_tracker.lua (hunk 1 above)
Comment (krawthekrow) [reply forward/testing/backend_tracker.lua:27]: The backend handling in Go is multithreaded, so it's necessary to wait for updates to propagate to Maglev whenever making changes to the backends before proceeding with the tests. I'm not sure why removing backends requires a longer sleep; possibly it takes some time to propagate channel closing in Go.

File: forward/testing/unit_tests.lua (hunk 2 above)
Comment (krawthekrow) [reply forward/testing/unit_tests.lua:54]: We can't differentiate between rewriting dropping packets when it's not supposed to and it taking a long time (unless we break the rewriting black box, or explicitly count engine breaths). Either of these would trigger that error. If we have longer tests in the future, we can always make the timeout configurable.

**Pull request merged**